src/vdaf.rs

```diff
@@ -91,7 +130,10 @@ pub trait Vdaf: Clone + Debug {
     type VerifyParam;
 
     /// An input share sent by a Client.
-    type InputShare: Clone + Debug;
+    type InputShare: Clone + Debug + Decode<Self::InputShareDecodingParameter> + Encode;
+
+    /// Contextual parameters needed for decoding messages into `Self::InputShare`.
+    type InputShareDecodingParameter;
```

Comment:
I think this associated type is redundant, namely because the aggregator is always going to know how to decode the input given its `VerifyParam`. I would drop this associated type and make add bound `InputShare` by `Decode<Self::VerifyParam>`.